Source/EasyNetQ/Producer/Rpc.cs

```diff
@@ -307,13 +306,7 @@ protected virtual void RequestPublish<TRequest>(TRequest request, string routing
             where TRequest : class
             where TResponse : class
         {
-            // HACK: I think we can live with this, because it will run only on exception, 
-            // it tries to preserve the default serialization behavior, 
-            // being able to also deserialize POCO objects that has constructors with parameters
-            // this avoids to introduce a custom class wrapper that will change the message payload
-            var body = JsonConvert.DeserializeObject<TResponse>(typeof(TResponse) == typeof(string) ? "''" : "{}");
-
-            var responseMessage = new Message<TResponse>(body);
+            var responseMessage = new Message<TResponse>(typeof(TResponse), default);
```

Comment:
Isn't backward compatibility broken here? An instance of the reference type was created here and stored in `Message<T>.Body`. Now `Message<T>.Body` will become `null`.

Reply:
It seems no, because `Message<T>.Body` was not used during deserialization of RPC's request.

Reply:
If you mean this [condition](https://github.com/EasyNetQ/EasyNetQ/blob/develop/Source/EasyNetQ/Producer/DefaultRpc.cs#L138), then I will agree that in this case `Message.Body` property itself is not used in EasyNetQ.

But nevertheless there remain 2 situations when such changes can lead to an error:
1. Overridden `RequestAsync` method with arbitrary user code. In my case it is. I was forced to override `RequestAsync` to be able to set a more complex timeout strategy. To be honest, I did not change the method `RegisterErrorHandling` but nothing prevents to call another implementation in which there will be logic tied to the `Body` property.
2. Applications connected to the RabbitMQ server written on other stacks/platforms - not EasyNetQ, not .NET at all.

Despite the problems described, it seems to me that it would be better to accept this change. At the very least, the original implementation would have been exactly like this without creating a dummy `TResponse` object for `Message.Body`.